kolibri/plugins/coach_tools/assets/src/state/getters.js

```diff
@@ -0,0 +1,109 @@
+
+const Constants = require('./constants');
+const logging = require('kolibri.lib.logging');
+
+/* given an array of objects sum the keys on those that pass the filter */
+function sumOfKeys(array, key, filter = () => true) {
+  return array
+    .filter(filter)
+    .reduce((prev, item) => prev + item[key], 0);
+}
+
+function countNodes(progressArray, filter) {
+  return sumOfKeys(progressArray, 'node_count', filter);
+}
+
+function calcProgress(progressArray, filter, count) {
+  const totalProgress = sumOfKeys(progressArray, 'total_progress', filter);
+  if (count) {
+    return totalProgress / count;
+  }
+  return undefined;
+}
+
+function onlyExercises(item) {
+  return item.kind === 'exercise';
+}
+
+function onlyContent(item) {
+  return item.kind !== 'exercise';
+}
+
+function genCompareFunc(sortColumn, sortOrder) {
+  const columnToKey = {};
+  columnToKey[Constants.TableColumns.NAME] = 'title';
+  columnToKey[Constants.TableColumns.EXERCISE] = 'exerciseProgress';
+  columnToKey[Constants.TableColumns.CONTENT] = 'contentProgress';
+  columnToKey[Constants.TableColumns.DATE] = 'lastActive';
+  const key = columnToKey[sortColumn];
+
+  // take into account sort order
+  const flipOrder = sortOrder === Constants.SortOrders.DESCENDING ? 1 : -1;
+  // default order of names is A-Z; everything else goes high-low
+  const flipNameCol = sortColumn !== Constants.TableColumns.NAME ? -1 : 1;
+  return (a, b) => {
+    if (a[key] > b[key]) { return 1 * flipOrder * flipNameCol; }
+    if (a[key] < b[key]) { return -1 * flipOrder * flipNameCol; }
+    return 0;
+  };
+}
+
+const getters = {
+  exerciseCount(state) {
+    return countNodes(state.pageState.content_scope_summary.progress, onlyExercises);
+  },
+  exerciseProgress(state) {
+    return calcProgress(
+      state.pageState.content_scope_summary.progress,
+      onlyExercises,
+      getters.exerciseCount(state)
+    );
+  },
+  contentCount(state) {
+    return countNodes(state.pageState.content_scope_summary.progress, onlyContent);
+  },
+  contentProgress(state) {
+    return calcProgress(
+      state.pageState.content_scope_summary.progress,
+      onlyContent,
+      getters.contentCount(state)
+    );
+  },
+  dataTable(state) {
+    let data = [];
+
+    // CONTENT or LEARNERS
+    if (state.pageState.view_by_content_or_learners === Constants.ViewBy.CONTENT) {
+      data = state.pageState.table_data.map(item => ({
+        kind: item.kind,
+        id: item.pk,
```

Comment:
This one is already a string - the other one is an integer.

Filing a related issue here: #657 - if we can standardize on strings, we can take out the `toString` logic.